rasterio/_base.pyx

```diff
@@ -930,7 +931,8 @@ cdef class DatasetBase:
         a, b, c, d, e, f, _, _, _ = self.transform
         width = self.width
         height = self.height
-        if b == d == 0:
+        eps = float_info.epsilon
+        if math.isclose(b, 0, abs_tol=eps) and math.isclose(d, 0, abs_tol=eps):
```

Comment:
Thoughts about passing in `float_info.epsilon` directly?
```suggestion
        if math.isclose(b, 0, abs_tol=float_info.epsilon) and math.isclose(d, 0, abs_tol=float_info.epsilon):
```

Reply:
I think I prefer the way that it is. I find it more easy to read.